scarb/tests/e2e/support/workspace_builder.rs

```diff
@@ -0,0 +1,48 @@
+use crate::support::project_builder::ProjectBuilder;
+use assert_fs::prelude::*;
+use toml_edit::{Array, Document, Item, Value};
+
+#[derive(Default)]
+pub struct WorkspaceBuilder {
+    members: Vec<String>,
+    package: Option<ProjectBuilder>,
+    manifest_extra: String,
+}
+
+impl WorkspaceBuilder {
+    pub fn start() -> Self {
+        Self::default()
+    }
+
+    pub fn add_member(mut self, member: impl Into<String>) -> Self {
+        self.members.push(member.into());
+        self
+    }
+
+    pub fn package(mut self, package: ProjectBuilder) -> Self {
+        self.package = Some(package);
+        self
+    }
```

Comment:
This seems like a kinda rare, single test, usage scenario. Can't we just use `extra(&str)` pattern that we have in ProjectBuilder? It should simplify much of this code as far as I see.

Reply:
Hmm, I am not sure about this one. 
Root packages can be tricky sometimes, as their location in workspace root can cause some weird edge-cases (e.g. our offline discussion about fmt), so naturally we may want more than one test case. 
Also, eradicating this field would make us to create package sources, pass the package manifest, sometimes also concatenated with other manifest extras on our own, which may be inconvenient at times.

Reply:
ok, convinced